modules/managed-identity/user-assigned-identity/README.md

```diff
@@ -0,0 +1,68 @@
+# User Assigned Identities
+
+This module deploys Microsoft.ManagedIdentity User Assigned Identities and optionally available children or extensions
+
+## Parameters
+
+| Name              | Type     | Required | Description                                                                                                                                                                                                                                                                                                                                                                                                    |
+| :---------------- | :------: | :------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `name`            | `string` | Yes      | Required. Name of the User Assigned Identity.                                                                                                                                                                                                                                                                                                                                                                  |
+| `location`        | `string` | No       | Optional. Location for all resources.                                                                                                                                                                                                                                                                                                                                                                          |
+| `lock`            | `string` | No       | Optional. Specify the type of lock.                                                                                                                                                                                                                                                                                                                                                                            |
+| `roleAssignments` | `array`  | No       | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11' |
```

Comment:
re: roleDefinitionIdOrName - a couple thoughts
1) the `name` of a roleDef is the guid or the name property on the resource. the `roleName` is the name of the property we use for the "displayName"
2) when supplying the `id` is there any reason we can't just ask for the guid/name?  i.e. why do we need the full resourceId (which is harder to work with)... 

IOW, the parameter becomes `roleDefinitionName` - to which we can describe was the `roleName` or `name` of the resource?